src/libcore/ptr/const_ptr.rs

```diff
@@ -295,6 +295,62 @@ impl<T: ?Sized> *const T {
         intrinsics::ptr_offset_from(self, origin)
     }
 
+    /// Returns whether two pointers are guaranteed equal. Returns `true` whenever the
+    /// pointers are known to be equal. If the pointers are equal, but the platform
+    /// isn't sure the pointers are equal, this function returns `false`.
```

Comment:
What is "the platform"?

I'd say something like

> Returns whether two pointers are definitely equal. In some evaluation contexts (e.g., compile-time evaluation),
> it is not always possible to determine equality of two pointers, so this function may spuriously return
> `false` for pointers that later actually turn out to be equal.

Reply:
I took your text and adjusted it a bit.